javascript/extractor/src/com/semmle/js/extractor/ASTExtractor.java

```diff
@@ -516,10 +516,97 @@ public Label visit(Literal nd, Context c) {
       String valueString = nd.getStringValue();
 
       trapwriter.addTuple("literals", valueString, source, key);
-      if (nd.isRegExp()) regexpExtractor.extract(source.substring(1, source.lastIndexOf('/')), nd);
+      if (nd.isRegExp()) {
+        OffsetTranslation offsets = new OffsetTranslation();
+        offsets.set(0, 1); // skip the initial '/'
+        regexpExtractor.extract(source.substring(1, source.lastIndexOf('/')), offsets, nd, false);
+      } else if (nd.isStringLiteral() && !c.isInsideType()) {
```

Comment:
Does this mean that we parse `"foo"` and `'bar'`, but not ``` `baz` ```? If not, then the `instanceof StringLiteral` bits in the QL parts of this PR  should be adjusted (unless we already landed the unification of template literals and string literals somewhere else).

Reply:
Constant template literals aren't parsed as regular expressions. There are few legitimate use-cases for the `StringLiteral` class that don't have the edge case of constant template literals to worry about, which is why I would like to extract constant template literals as string literals instead of special-casing it at the use-sites. You and Max both expressed dislike for that idea so I left it out of this PR to avoid getting stuck on that issue, though I still think it's the right way forward.